src/Behat/Testwork/ServiceContainer/ExtensionManager.php

```diff
@@ -175,20 +174,12 @@ private function initialize($locator)
      */
     private function instantiateExtension($locator)
     {
-        if (class_exists($class = $locator)) {
-            return new $class;
-        }
-
-        if (class_exists($class = $this->getFullExtensionClass($locator))) {
-            return new $class;
-        }
-
-        if (file_exists($locator)) {
-            return require($locator);
-        }
-
-        if (file_exists($path = $this->extensionsPath . DIRECTORY_SEPARATOR . $locator)) {
-            return require($path);
+        foreach ($this->instantiators as $instantiator) {
+            try {
+                return $instantiator->instantiate($locator);
+            } catch (ExtensionInitializationException $e) {
+                // ignored if the instantiator does not support the locator
+            }
         }
 
         throw new ExtensionInitializationException(sprintf(
```

Comment:
Not sure if building the last exception from the other exceptions is relevant, as these are just "failures". Removing the `class or file` part looks like a good idea, to stay generic.

What could be done would be to list which instantiators were tried ?